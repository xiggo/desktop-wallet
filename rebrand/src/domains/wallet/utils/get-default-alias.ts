import { Networks } from "@payvo/sdk";
import { Contracts } from "@payvo/sdk-profiles";

interface GetDefaultAliasInput {
	profile: Contracts.IProfile;
	network: Networks.Network;
}

export const getDefaultAlias = ({ profile, network }: GetDefaultAliasInput): string => {
	const makeAlias = (count: number) => `${network.displayName()} #${count}`;

	const sameCoinWallets = profile.wallets().findByCoinWithNetwork(network.coin(), network.id());

	let counter = sameCoinWallets.length;

	if (counter === 0) {
		counter = 1;
	}

	while (profile.wallets().findByAlias(makeAlias(counter))) {
		counter++;
	}

	return makeAlias(counter);
};
