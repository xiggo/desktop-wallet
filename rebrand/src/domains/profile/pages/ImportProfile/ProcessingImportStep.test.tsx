import fs from "fs";
import { Profile } from "@payvo/sdk-profiles";
import userEvent from "@testing-library/user-event";
import React from "react";

import { ProcessingImport } from "@/domains/profile/pages/ImportProfile/ProcessingImportStep";
import { env, render, screen, waitFor } from "@/utils/testing-library";

let dwe: any;
let passwordProtectedDwe: any;
let json: any;
let jsonEmpty: any;

const submitID = "PasswordModal__submit-button";

describe("Import Profile - Processing import", () => {
	beforeAll(() => {
		const jsonEmptyContent = fs.readFileSync("src/tests/fixtures/profile/import/d2_test_wallets-empty.json");
		const jsonContent = fs.readFileSync("src/tests/fixtures/profile/import/d2_test_wallets.json");
		const dweFileContents = fs.readFileSync("src/tests/fixtures/profile/import/profile.dwe");
		const passwordProtectedDweFileContents = fs.readFileSync(
			"src/tests/fixtures/profile/import/password-protected-profile.dwe",
		);

		dwe = { content: dweFileContents.toString(), extension: ".dwe", name: "profile.dwe" };
		passwordProtectedDwe = {
			content: passwordProtectedDweFileContents.toString(),
			extension: ".dwe",
			name: "export",
		};

		json = { content: jsonContent.toString(), extension: ".json", name: "export" };
		jsonEmpty = { content: jsonEmptyContent.toString(), extension: ".json", name: "export" };
	});

	it("should not run import process if file is not provided", () => {
		const { container } = render(<ProcessingImport env={env} />);

		expect(container).toMatchSnapshot();
	});

	it("should successfully import dwe profile", async () => {
		const onSuccess = jest.fn();
		const { container } = render(<ProcessingImport env={env} file={dwe} onSuccess={onSuccess} />);
		await waitFor(() => expect(onSuccess).toHaveBeenCalledWith(expect.any(Profile)));

		expect(container).toMatchSnapshot();
	});

	it("should successfully import json profile", async () => {
		const onSuccess = jest.fn();
		const { container } = render(<ProcessingImport env={env} file={json} onSuccess={onSuccess} />);
		await waitFor(() => expect(onSuccess).toHaveBeenCalledWith(expect.any(Profile)));

		expect(container).toMatchSnapshot();
	});

	it("should require password for password-protected profile import", async () => {
		const onPasswordChange = jest.fn();
		const { container } = render(
			<ProcessingImport env={env} file={passwordProtectedDwe} onPasswordChange={onPasswordChange} />,
		);

		await expect(screen.findByTestId("modal__inner")).resolves.toBeVisible();

		userEvent.paste(screen.getByTestId("PasswordModal__input"), "S3cUrePa$sword");

		await expect(screen.findByTestId(submitID)).resolves.toBeVisible();

		userEvent.click(screen.getByTestId(submitID));

		await waitFor(() => expect(onPasswordChange).toHaveBeenCalledWith("S3cUrePa$sword"));

		expect(container).toMatchSnapshot();
	});

	it("should call back when password modal is closed", async () => {
		const onBack = jest.fn();
		render(<ProcessingImport env={env} file={passwordProtectedDwe} onBack={onBack} password="test" />);

		await expect(screen.findByTestId("modal__inner")).resolves.toBeVisible();

		userEvent.click(screen.getByTestId("modal__close-btn"));

		await waitFor(() => expect(onBack).toHaveBeenCalledWith());
	});

	it("should handle invalid password", async () => {
		const onPasswordChange = jest.fn();
		const { container } = render(
			<ProcessingImport
				env={env}
				file={passwordProtectedDwe}
				onPasswordChange={onPasswordChange}
				password="test"
			/>,
		);

		await expect(screen.findByTestId("modal__inner")).resolves.toBeVisible();

		userEvent.paste(screen.getByTestId("PasswordModal__input"), "invalid");

		await expect(screen.findByTestId(submitID)).resolves.toBeVisible();

		userEvent.click(screen.getByTestId(submitID));

		await expect(screen.findByTestId("modal__inner")).resolves.toBeVisible();

		await waitFor(() => expect(screen.getByTestId("PasswordModal__input")).toHaveValue("invalid"));
		await waitFor(() => expect(onPasswordChange).not.toHaveBeenCalledWith("testtest2"));

		expect(container).toMatchSnapshot();
	});

	it("should enter password again", async () => {
		const onPasswordChange = jest.fn();
		const { container } = render(
			<ProcessingImport
				env={env}
				file={passwordProtectedDwe}
				onPasswordChange={onPasswordChange}
				password="test"
				shouldRequestPassword
			/>,
		);

		await expect(screen.findByTestId("modal__inner")).resolves.toBeVisible();

		userEvent.paste(screen.getByTestId("PasswordModal__input"), "invalid");

		await expect(screen.findByTestId(submitID)).resolves.toBeVisible();

		userEvent.click(screen.getByTestId(submitID));

		await expect(screen.findByTestId("modal__inner")).resolves.toBeVisible();

		await waitFor(() => expect(onPasswordChange).not.toHaveBeenCalledWith("testtest2"));

		expect(container).toMatchSnapshot();
	});

	it("should show error if json import has an error", async () => {
		const onError = jest.fn();
		const { container } = render(<ProcessingImport env={env} file={jsonEmpty} onError={onError} password="test" />);

		await waitFor(() => expect(onError).toHaveBeenCalledWith("MissingWallets"));

		expect(container).toMatchSnapshot();
	});

	it("should handle import error", async () => {
		const onError = jest.fn();
		const { container } = render(
			<ProcessingImport
				env={env}
				file={{ content: "corrupted format", extension: ".dwe", name: "test.dwe" }}
				onError={onError}
				password="test"
			/>,
		);

		await waitFor(() =>
			expect(onError).toHaveBeenCalledWith(expect.stringMatching(/Failed to decode or decrypt the profile/)),
		);

		expect(container).toMatchSnapshot();
	});
});
