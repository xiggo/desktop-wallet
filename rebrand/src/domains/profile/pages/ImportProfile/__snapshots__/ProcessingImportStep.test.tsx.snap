// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Import Profile - Processing import should enter password again 1`] = `
<div>
  <div
    class="mx-auto max-w-xl"
    data-testid="ProcessingImport"
  >
    <div
      class="flex justify-between items-end bg-theme-background"
    >
      <div
        class="space-y-4"
      >
        <h1
          class="mb-0 text-2xl"
          data-testid="header__title"
        >
          Import Profile
        </h1>
        <div
          class="text-theme-secondary-text"
          data-testid="header__subtitle"
        >
          The file with name export is being imported.
        </div>
      </div>
    </div>
    <div
      class="text-theme-secondary-text"
    >
      Please wait...
    </div>
    <div
      class="mt-8"
    >
      <div
        class="p-4 rounded-lg border-2 border-theme-secondary-200 dark:border-theme-secondary-800"
      >
        <div
          class="flex justify-between items-center space-x-4"
        >
          <div
            class="flex items-center space-x-4"
          >
            <div
              class="sc-bdvvtL kCgesw"
              height="40"
              width="40"
            >
              <svg>
                extension-dwe.svg
              </svg>
            </div>
            <div
              class="font-semibold"
            >
              <span
                class="TruncateMiddle__Wrapper-sc-ue8va6-0"
                data-testid="TruncateMiddle"
              >
                export
              </span>
            </div>
          </div>
          <div
            class="Spinner-sc-1rpx65r-0 lfdpzU"
            color="info"
          />
        </div>
      </div>
    </div>
    <div
      class="text-left items-left"
    >
      <div
        class="flex fixed inset-0 z-50 py-20 w-full h-full bg-opacity-60 dark:bg-opacity-80 bg-theme-secondary-900-rgba overflow-overlay dark:bg-black-rgba"
        data-testid="modal__overlay"
      >
        <div
          class="Modal__ModalContainer-sc-5lu2mc-0 fVxcFR"
          tabindex="-1"
        >
          <div
            class="flex overflow-hidden relative flex-col p-10 shadow-2xl rounded-2.5xl bg-theme-background"
            data-testid="modal__inner"
          >
            <div
              class="absolute top-0 right-0 z-10 mt-5 mr-5 rounded transition-all duration-100 ease-linear bg-theme-primary-100 dark:bg-theme-secondary-800 dark:text-theme-secondary-600 dark:hover:bg-theme-secondary-700 dark:hover:text-theme-secondary-400 hover:bg-theme-primary-300"
            >
              <button
                class="OriginalButton__StyledButton-sc-o0fs2j-0 enzTPF w-11 h-11"
                data-testid="modal__close-btn"
                type="button"
              >
                <div
                  class="flex items-center space-x-2"
                >
                  <div
                    class="sc-bdvvtL fCrhkg"
                    height="16"
                    width="16"
                  >
                    <svg>
                      cross.svg
                    </svg>
                  </div>
                </div>
              </button>
            </div>
            <div
              class="relative space-y-4"
            >
              <h2
                class="mb-0 text-2xl font-bold items-left"
              >
                Profile Password
              </h2>
              <div
                class="flex-1"
              >
                <div
                  class="whitespace-pre-line text-theme-secondary-text"
                >
                  The profile you are importing contains a password that has been set for security purposes. Enter the password to confirm.
                </div>
                <form
                  class="mt-8"
                  data-testid="Form"
                >
                  <fieldset
                    class="FormField__FormFieldStyled-sc-q4bblx-0 dFNJMk flex flex-col space-y-2"
                  >
                    <label
                      class="flex mb-2 text-sm font-semibold transition-colors duration-100 FormLabel text-theme-secondary-text"
                      data-testid="FormLabel"
                      for="password"
                    >
                      Password
                    </label>
                    <div
                      class="Input__InputWrapperStyled-sc-1i8sd1r-0 cLNVAR"
                    >
                      <div
                        class="relative flex flex-1 h-full"
                      >
                        <input
                          class="Input__InputStyled-sc-1i8sd1r-1 hPntwa p-0 border-none bg-transparent focus:ring-0 no-ligatures w-full"
                          data-testid="PasswordModal__input"
                          name="password"
                          type="password"
                          value=""
                        />
                      </div>
                      <div
                        class="flex items-center space-x-3 divide-x divide-theme-secondary-300 dark:divide-theme-secondary-800 text-theme-primary-300 dark:text-theme-secondary-600"
                        data-testid="Input__addon-end"
                      >
                        <div
                          class=""
                        >
                          <button
                            class="flex relative justify-center items-center w-full h-full text-2xl focus:outline-none ring-focus"
                            data-ring-focus-margin="-m-1"
                            data-testid="InputPassword__toggle"
                            type="button"
                          >
                            <div
                              class="sc-bdvvtL qGXhc"
                              height="20"
                              width="20"
                            >
                              <svg>
                                eye.svg
                              </svg>
                            </div>
                          </button>
                        </div>
                      </div>
                    </div>
                  </fieldset>
                  <div
                    class="flex justify-end mt-8 space-x-3"
                  >
                    <button
                      class="OriginalButton__StyledButton-sc-o0fs2j-0 isLyLB"
                      data-testid="PasswordModal__submit-button"
                      type="submit"
                    >
                      <div
                        class="flex items-center space-x-2"
                      >
                        Confirm
                      </div>
                    </button>
                  </div>
                </form>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Import Profile - Processing import should handle import error 1`] = `
<div>
  <div
    class="mx-auto max-w-xl"
    data-testid="ProcessingImport"
  >
    <div
      class="flex justify-between items-end bg-theme-background"
    >
      <div
        class="space-y-4"
      >
        <h1
          class="mb-0 text-2xl"
          data-testid="header__title"
        >
          Import Profile
        </h1>
        <div
          class="text-theme-secondary-text"
          data-testid="header__subtitle"
        >
          The file with name test.dwe is being imported.
        </div>
      </div>
    </div>
    <div
      class="text-theme-secondary-text"
    >
      Please wait...
    </div>
    <div
      class="mt-8"
    >
      <div
        class="p-4 rounded-lg border-2 border-theme-secondary-200 dark:border-theme-secondary-800"
      >
        <div
          class="flex justify-between items-center space-x-4"
        >
          <div
            class="flex items-center space-x-4"
          >
            <div
              class="sc-bdvvtL kCgesw"
              height="40"
              width="40"
            >
              <svg>
                extension-dwe.svg
              </svg>
            </div>
            <div
              class="font-semibold"
            >
              <span
                class="TruncateMiddle__Wrapper-sc-ue8va6-0"
                data-testid="TruncateMiddle"
              >
                test.dwe
              </span>
            </div>
          </div>
          <div
            class="Spinner-sc-1rpx65r-0 lfdpzU"
            color="info"
          />
        </div>
      </div>
    </div>
    <div
      class="text-left items-left"
    />
  </div>
</div>
`;

exports[`Import Profile - Processing import should handle invalid password 1`] = `
<div>
  <div
    class="mx-auto max-w-xl"
    data-testid="ProcessingImport"
  >
    <div
      class="flex justify-between items-end bg-theme-background"
    >
      <div
        class="space-y-4"
      >
        <h1
          class="mb-0 text-2xl"
          data-testid="header__title"
        >
          Import Profile
        </h1>
        <div
          class="text-theme-secondary-text"
          data-testid="header__subtitle"
        >
          The file with name export is being imported.
        </div>
      </div>
    </div>
    <div
      class="text-theme-secondary-text"
    >
      Please wait...
    </div>
    <div
      class="mt-8"
    >
      <div
        class="p-4 rounded-lg border-2 border-theme-secondary-200 dark:border-theme-secondary-800"
      >
        <div
          class="flex justify-between items-center space-x-4"
        >
          <div
            class="flex items-center space-x-4"
          >
            <div
              class="sc-bdvvtL kCgesw"
              height="40"
              width="40"
            >
              <svg>
                extension-dwe.svg
              </svg>
            </div>
            <div
              class="font-semibold"
            >
              <span
                class="TruncateMiddle__Wrapper-sc-ue8va6-0"
                data-testid="TruncateMiddle"
              >
                export
              </span>
            </div>
          </div>
          <div
            class="Spinner-sc-1rpx65r-0 lfdpzU"
            color="info"
          />
        </div>
      </div>
    </div>
    <div
      class="text-left items-left"
    >
      <div
        class="flex fixed inset-0 z-50 py-20 w-full h-full bg-opacity-60 dark:bg-opacity-80 bg-theme-secondary-900-rgba overflow-overlay dark:bg-black-rgba"
        data-testid="modal__overlay"
      >
        <div
          class="Modal__ModalContainer-sc-5lu2mc-0 fVxcFR"
          tabindex="-1"
        >
          <div
            class="flex overflow-hidden relative flex-col p-10 shadow-2xl rounded-2.5xl bg-theme-background"
            data-testid="modal__inner"
          >
            <div
              class="absolute top-0 right-0 z-10 mt-5 mr-5 rounded transition-all duration-100 ease-linear bg-theme-primary-100 dark:bg-theme-secondary-800 dark:text-theme-secondary-600 dark:hover:bg-theme-secondary-700 dark:hover:text-theme-secondary-400 hover:bg-theme-primary-300"
            >
              <button
                class="OriginalButton__StyledButton-sc-o0fs2j-0 enzTPF w-11 h-11"
                data-testid="modal__close-btn"
                type="button"
              >
                <div
                  class="flex items-center space-x-2"
                >
                  <div
                    class="sc-bdvvtL fCrhkg"
                    height="16"
                    width="16"
                  >
                    <svg>
                      cross.svg
                    </svg>
                  </div>
                </div>
              </button>
            </div>
            <div
              class="relative space-y-4"
            >
              <h2
                class="mb-0 text-2xl font-bold items-left"
              >
                Profile Password
              </h2>
              <div
                class="flex-1"
              >
                <div
                  class="whitespace-pre-line text-theme-secondary-text"
                >
                  The profile you are importing contains a password that has been set for security purposes. Enter the password to confirm.
                </div>
                <form
                  class="mt-8"
                  data-testid="Form"
                >
                  <fieldset
                    class="FormField__FormFieldStyled-sc-q4bblx-0 dFNJMk flex flex-col space-y-2"
                  >
                    <label
                      class="flex mb-2 text-sm font-semibold transition-colors duration-100 FormLabel text-theme-secondary-text"
                      data-testid="FormLabel"
                      for="password"
                    >
                      Password
                    </label>
                    <div
                      class="Input__InputWrapperStyled-sc-1i8sd1r-0 cLNVAR"
                    >
                      <div
                        class="relative flex flex-1 h-full"
                      >
                        <input
                          class="Input__InputStyled-sc-1i8sd1r-1 hPntwa p-0 border-none bg-transparent focus:ring-0 no-ligatures w-full"
                          data-testid="PasswordModal__input"
                          name="password"
                          type="password"
                          value=""
                        />
                      </div>
                      <div
                        class="flex items-center space-x-3 divide-x divide-theme-secondary-300 dark:divide-theme-secondary-800 text-theme-primary-300 dark:text-theme-secondary-600"
                        data-testid="Input__addon-end"
                      >
                        <div
                          class=""
                        >
                          <button
                            class="flex relative justify-center items-center w-full h-full text-2xl focus:outline-none ring-focus"
                            data-ring-focus-margin="-m-1"
                            data-testid="InputPassword__toggle"
                            type="button"
                          >
                            <div
                              class="sc-bdvvtL qGXhc"
                              height="20"
                              width="20"
                            >
                              <svg>
                                eye.svg
                              </svg>
                            </div>
                          </button>
                        </div>
                      </div>
                    </div>
                  </fieldset>
                  <div
                    class="flex justify-end mt-8 space-x-3"
                  >
                    <button
                      class="OriginalButton__StyledButton-sc-o0fs2j-0 isLyLB"
                      data-testid="PasswordModal__submit-button"
                      type="submit"
                    >
                      <div
                        class="flex items-center space-x-2"
                      >
                        Confirm
                      </div>
                    </button>
                  </div>
                </form>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Import Profile - Processing import should not run import process if file is not provided 1`] = `
<div>
  <div
    class="mx-auto max-w-xl"
    data-testid="ProcessingImport"
  >
    <div
      class="flex justify-between items-end bg-theme-background"
    >
      <div
        class="space-y-4"
      >
        <h1
          class="mb-0 text-2xl"
          data-testid="header__title"
        >
          Import Profile
        </h1>
        <div
          class="text-theme-secondary-text"
          data-testid="header__subtitle"
        >
          The file with name  is being imported.
        </div>
      </div>
    </div>
    <div
      class="text-theme-secondary-text"
    >
      Please wait...
    </div>
    <div
      class="mt-8"
    />
    <div
      class="text-left items-left"
    />
  </div>
</div>
`;

exports[`Import Profile - Processing import should require password for password-protected profile import 1`] = `
<div>
  <div
    class="mx-auto max-w-xl"
    data-testid="ProcessingImport"
  >
    <div
      class="flex justify-between items-end bg-theme-background"
    >
      <div
        class="space-y-4"
      >
        <h1
          class="mb-0 text-2xl"
          data-testid="header__title"
        >
          Import Profile
        </h1>
        <div
          class="text-theme-secondary-text"
          data-testid="header__subtitle"
        >
          The file with name export is being imported.
        </div>
      </div>
    </div>
    <div
      class="text-theme-secondary-text"
    >
      Please wait...
    </div>
    <div
      class="mt-8"
    >
      <div
        class="p-4 rounded-lg border-2 border-theme-secondary-200 dark:border-theme-secondary-800"
      >
        <div
          class="flex justify-between items-center space-x-4"
        >
          <div
            class="flex items-center space-x-4"
          >
            <div
              class="sc-bdvvtL kCgesw"
              height="40"
              width="40"
            >
              <svg>
                extension-dwe.svg
              </svg>
            </div>
            <div
              class="font-semibold"
            >
              <span
                class="TruncateMiddle__Wrapper-sc-ue8va6-0"
                data-testid="TruncateMiddle"
              >
                export
              </span>
            </div>
          </div>
          <div
            class="Spinner-sc-1rpx65r-0 lfdpzU"
            color="info"
          />
        </div>
      </div>
    </div>
    <div
      class="text-left items-left"
    >
      <div
        class="flex fixed inset-0 z-50 py-20 w-full h-full bg-opacity-60 dark:bg-opacity-80 bg-theme-secondary-900-rgba overflow-overlay dark:bg-black-rgba"
        data-testid="modal__overlay"
      >
        <div
          class="Modal__ModalContainer-sc-5lu2mc-0 fVxcFR"
          tabindex="-1"
        >
          <div
            class="flex overflow-hidden relative flex-col p-10 shadow-2xl rounded-2.5xl bg-theme-background"
            data-testid="modal__inner"
          >
            <div
              class="absolute top-0 right-0 z-10 mt-5 mr-5 rounded transition-all duration-100 ease-linear bg-theme-primary-100 dark:bg-theme-secondary-800 dark:text-theme-secondary-600 dark:hover:bg-theme-secondary-700 dark:hover:text-theme-secondary-400 hover:bg-theme-primary-300"
            >
              <button
                class="OriginalButton__StyledButton-sc-o0fs2j-0 enzTPF w-11 h-11"
                data-testid="modal__close-btn"
                type="button"
              >
                <div
                  class="flex items-center space-x-2"
                >
                  <div
                    class="sc-bdvvtL fCrhkg"
                    height="16"
                    width="16"
                  >
                    <svg>
                      cross.svg
                    </svg>
                  </div>
                </div>
              </button>
            </div>
            <div
              class="relative space-y-4"
            >
              <h2
                class="mb-0 text-2xl font-bold items-left"
              >
                Profile Password
              </h2>
              <div
                class="flex-1"
              >
                <div
                  class="whitespace-pre-line text-theme-secondary-text"
                >
                  The profile you are importing contains a password that has been set for security purposes. Enter the password to confirm.
                </div>
                <form
                  class="mt-8"
                  data-testid="Form"
                >
                  <fieldset
                    class="FormField__FormFieldStyled-sc-q4bblx-0 dFNJMk flex flex-col space-y-2"
                  >
                    <label
                      class="flex mb-2 text-sm font-semibold transition-colors duration-100 FormLabel text-theme-secondary-text"
                      data-testid="FormLabel"
                      for="password"
                    >
                      Password
                    </label>
                    <div
                      class="Input__InputWrapperStyled-sc-1i8sd1r-0 cLNVAR"
                    >
                      <div
                        class="relative flex flex-1 h-full"
                      >
                        <input
                          class="Input__InputStyled-sc-1i8sd1r-1 hPntwa p-0 border-none bg-transparent focus:ring-0 no-ligatures w-full"
                          data-testid="PasswordModal__input"
                          name="password"
                          type="password"
                          value=""
                        />
                      </div>
                      <div
                        class="flex items-center space-x-3 divide-x divide-theme-secondary-300 dark:divide-theme-secondary-800 text-theme-primary-300 dark:text-theme-secondary-600"
                        data-testid="Input__addon-end"
                      >
                        <div
                          class=""
                        >
                          <button
                            class="flex relative justify-center items-center w-full h-full text-2xl focus:outline-none ring-focus"
                            data-ring-focus-margin="-m-1"
                            data-testid="InputPassword__toggle"
                            type="button"
                          >
                            <div
                              class="sc-bdvvtL qGXhc"
                              height="20"
                              width="20"
                            >
                              <svg>
                                eye.svg
                              </svg>
                            </div>
                          </button>
                        </div>
                      </div>
                    </div>
                  </fieldset>
                  <div
                    class="flex justify-end mt-8 space-x-3"
                  >
                    <button
                      class="OriginalButton__StyledButton-sc-o0fs2j-0 isLyLB"
                      data-testid="PasswordModal__submit-button"
                      type="submit"
                    >
                      <div
                        class="flex items-center space-x-2"
                      >
                        Confirm
                      </div>
                    </button>
                  </div>
                </form>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Import Profile - Processing import should show error if json import has an error 1`] = `
<div>
  <div
    class="mx-auto max-w-xl"
    data-testid="ProcessingImport"
  >
    <div
      class="flex justify-between items-end bg-theme-background"
    >
      <div
        class="space-y-4"
      >
        <h1
          class="mb-0 text-2xl"
          data-testid="header__title"
        >
          Import Profile
        </h1>
        <div
          class="text-theme-secondary-text"
          data-testid="header__subtitle"
        >
          The file with name export is being imported.
        </div>
      </div>
    </div>
    <div
      class="text-theme-secondary-text"
    >
      Please wait...
    </div>
    <div
      class="mt-8"
    >
      <div
        class="p-4 rounded-lg border-2 border-theme-secondary-200 dark:border-theme-secondary-800"
      >
        <div
          class="flex justify-between items-center space-x-4"
        >
          <div
            class="flex items-center space-x-4"
          >
            <div
              class="sc-bdvvtL kCgesw"
              height="40"
              width="40"
            >
              <svg>
                extension-json.svg
              </svg>
            </div>
            <div
              class="font-semibold"
            >
              <span
                class="TruncateMiddle__Wrapper-sc-ue8va6-0"
                data-testid="TruncateMiddle"
              >
                export
              </span>
            </div>
          </div>
          <div
            class="Spinner-sc-1rpx65r-0 lfdpzU"
            color="info"
          />
        </div>
      </div>
    </div>
    <div
      class="text-left items-left"
    />
  </div>
</div>
`;

exports[`Import Profile - Processing import should successfully import dwe profile 1`] = `
<div>
  <div
    class="mx-auto max-w-xl"
    data-testid="ProcessingImport"
  >
    <div
      class="flex justify-between items-end bg-theme-background"
    >
      <div
        class="space-y-4"
      >
        <h1
          class="mb-0 text-2xl"
          data-testid="header__title"
        >
          Import Profile
        </h1>
        <div
          class="text-theme-secondary-text"
          data-testid="header__subtitle"
        >
          The file with name profile.dwe is being imported.
        </div>
      </div>
    </div>
    <div
      class="text-theme-secondary-text"
    >
      Please wait...
    </div>
    <div
      class="mt-8"
    >
      <div
        class="p-4 rounded-lg border-2 border-theme-secondary-200 dark:border-theme-secondary-800"
      >
        <div
          class="flex justify-between items-center space-x-4"
        >
          <div
            class="flex items-center space-x-4"
          >
            <div
              class="sc-bdvvtL kCgesw"
              height="40"
              width="40"
            >
              <svg>
                extension-dwe.svg
              </svg>
            </div>
            <div
              class="font-semibold"
            >
              <span
                class="TruncateMiddle__Wrapper-sc-ue8va6-0"
                data-testid="TruncateMiddle"
              >
                profile.dwe
              </span>
            </div>
          </div>
          <div
            class="Spinner-sc-1rpx65r-0 lfdpzU"
            color="info"
          />
        </div>
      </div>
    </div>
    <div
      class="text-left items-left"
    />
  </div>
</div>
`;

exports[`Import Profile - Processing import should successfully import json profile 1`] = `
<div>
  <div
    class="mx-auto max-w-xl"
    data-testid="ProcessingImport"
  >
    <div
      class="flex justify-between items-end bg-theme-background"
    >
      <div
        class="space-y-4"
      >
        <h1
          class="mb-0 text-2xl"
          data-testid="header__title"
        >
          Import Profile
        </h1>
        <div
          class="text-theme-secondary-text"
          data-testid="header__subtitle"
        >
          The file with name export is being imported.
        </div>
      </div>
    </div>
    <div
      class="text-theme-secondary-text"
    >
      Please wait...
    </div>
    <div
      class="mt-8"
    >
      <div
        class="p-4 rounded-lg border-2 border-theme-secondary-200 dark:border-theme-secondary-800"
      >
        <div
          class="flex justify-between items-center space-x-4"
        >
          <div
            class="flex items-center space-x-4"
          >
            <div
              class="sc-bdvvtL kCgesw"
              height="40"
              width="40"
            >
              <svg>
                extension-json.svg
              </svg>
            </div>
            <div
              class="font-semibold"
            >
              <span
                class="TruncateMiddle__Wrapper-sc-ue8va6-0"
                data-testid="TruncateMiddle"
              >
                export
              </span>
            </div>
          </div>
          <div
            class="Spinner-sc-1rpx65r-0 lfdpzU"
            color="info"
          />
        </div>
      </div>
    </div>
    <div
      class="text-left items-left"
    />
  </div>
</div>
`;
